src/apps/app_search.erl

```diff
@@ -41,6 +41,22 @@ add_entry(undefined, _, _, _) -> do_nothing;
 add_entry(PID, Name, Value, ID) ->
 	PID ! {add_tx, Name, Value, ID}.
 
+get_tags_by_id(TXID) ->
+	get_tags_by_id(whereis(http_search_node), TXID).
+get_tags_by_id(PID, TXID) ->
+	PID ! {get_tags, TXID, self()},
+	receive Tags ->
+		lists:map(
+			fun(Tag) ->
+				{_, Name, Value, _} = Tag,
```

Comment:
I don't know what the first and the forth element is here, but I'm guessing it would make more sense to do this transformation in the other process, where there is a context about where this tuple is from.